components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/utils/BPMNDataReceiverConfig.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.bpmn.analytics.publisher.utils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.bpmn.analytics.publisher.AnalyticsPublisherConstants;
+import org.wso2.carbon.bpmn.analytics.publisher.internal.BPMNAnalyticsHolder;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.api.Resource;
+import org.wso2.carbon.user.api.UserStoreException;
+
+/**
+ * BPMNDataReceiverConfig is used by AnalyticsPublisher to retrieve user name and password
+ */
+public class BPMNDataReceiverConfig {
+    private static final Log log = LogFactory.getLog(BPMNDataReceiverConfig.class);
+
+    /**
+     * Get thrift url of data receiver
+     *
+     * @return thrift url of data receiver
+     * @throws RegistryException
+     */
+    public static String getThriftURL() throws RegistryException {
```

Comment:
A better design of exception handling would be, create exception classes for your own domain/context. For example, for the user of the class BPMNDataReceiverConfig.getThriftURL, he will not care if it's an underlying registry error that caused it. But rather, it should be wrapped by your own maybe BPMNConfigurationException or something like that. The idea is, each layer should handle their own specific logic, and not to expose unneeded information to the outer layers.